server/portal/apps/auth/views.py

```diff
@@ -54,6 +58,26 @@ def agave_oauth(request):
     return HttpResponseRedirect(authorization_url)
 
 
+def launch_setup_checks(user):
+    """Perform any onboarding checks that may spawn celery tasks
+    """
+
+    # Check onboarding settings
+    new_user_setup_check(user)
+    if not user.profile.setup_complete:
+        logger.info("Executing onboarding setup steps for %s", user.username)
+        execute_setup_steps.apply_async(args=[user.username])
+
+    # Apply asynchronous long calls
+    logger.info("Starting system setup celery tasks for {username}".format(username=user.username))
+    index_allocations.apply_async(args=[user.username])
+
+    system_names = get_user_storage_systems(user.username, settings.PORTAL_DATA_DEPOT_LOCAL_STORAGE_SYSTEMS)
+    group(
+        setup_user.s(user.username, system) for system in system_names
+    ).apply_async()
+
```

Comment:
the six lines above have the bug (unrelated to this PR) described here https://jira.tacc.utexas.edu/browse/FP-643 under "Problem B".  I think I'll pull the "Problem B" in its own ticket and put it under this epic.  If there is no objection, I'll do that.  So let me know how we want to track that one.

[my thought process is either we should do it in this PR or we should create a ticket.  I prefer ticket to keep this PR from getting bigger.  but fine either way]

Reply:
None of this stuff should be run in parallel if there are going to be race conditions - that's why we have the onboarding system run everything sequentially. Problem B exists because we didn't have sequential, asynchronous onboarding at the time. If we wanted to solve both problems, we would do sequential onboarding steps to retrieve allocations and not run system creation until after the allocations are retrieved. The entire time, the onboarding system would direct the user to the "wait" page while this happens and provide status updates.

Reply:
yes, i'm just checking to see if we have a ticket for this follow up work.